ios/MaasCore/Sources/MaasCore/Api/ApiConfiguration.swift

```diff
@@ -1,30 +1,16 @@
-class ApiConfiguration: ApiConfig {
-    
-    private init() {}
+public protocol ApiConfig: class {
+    var baseUrl: String { get }
     
-    static let shared = ApiConfiguration()
-    static var config: CoreBinary.ApiConfiguration { shared.config }
-
-    var baseUrl: String { "" }
-    var apiKey: String { "" }
+    ///  - Tag: ApiConfig.apiKey
+    var apiKey: String { get }
     
-    func getIdToken() -> String? {
-        return ""
-    }
+    ///  - Tag: ApiConfig.getIdToken
+    func getIdToken() -> String?
     
-    var logger: Logger? { ApiLogger.default }
-}
+    /// - Tag: ApiConfig.getIdToken
+    ///  must ensure that when completion is called, [getIdToken()](x-source-tag://ApiConfig.getIdToken) will return refreshed token
+    func refreshIdToken(completion: @escaping (String?) -> ())
```

Comment:
I think it's a sensible choice to have a global way to configure `getIdToken` and `refreshIdToken` 👍 In the future, if we want to support more unauthenticated use cases (transit, route search, etc), we may make these optional.